src/snowflake/snowpark/modin/plugin/__init__.py

```diff
@@ -26,5 +37,26 @@
 supported_modin_version = "0.28.1"
 if version.parse(modin.__version__) != version.parse(supported_modin_version):
     raise ImportError(
-        "Installed Modin version is not supported. " + install_msg
+        f"The Modin version installed ({modin.__version__}) does not match the supported Modin version in"
+        + f" Snowpark pandas ({supported_modin_version}). "
+        + install_msg
     )  # pragma: no cover
+
+
+warnings.warn(
+    "Snowpark pandas is currently in Public Preview."
+    + " See https://docs.snowflake.com/LIMITEDACCESS/snowpark-pandas for details.",
+    stacklevel=1,
```

Comment:
```suggestion
```

nit: `stacklevel=1` is the default

Reply:
Not specifying stacklevel is a flake8 warning:
```
src/snowflake/snowpark/modin/plugin/__init__.py:46:1: B028 No explicit stacklevel keyword argument found. The warn method from the warnings module uses a stacklevel of 1 by default. This will only show a stack trace for the line on which the warn method is called. It is therefore recommended to use a stacklevel of 2 or greater to provide more information to the user.
```

Reply:
I don't understand how `stacklevel` works; do you? stacklevel 1, 2, 0, 3, all do the same thing here:

```python
import warnings

warnings.filterwarnings('error', message='.*evie.*')

def do_deprecated():
    deprecated_api()
def deprecated_api():
    warnings.warn('preview', DeprecationWarning, stacklevel=0)
do_deprecated()
```


![Screenshot 2024-04-24 at 4 53 54 PM](https://github.com/snowflakedb/snowpark-python/assets/149194323/d3c2a048-f1af-4507-8387-c54baa2c3423)

Reply:
I tried your example, and removing the `filterwarnigns` call makes the `stacklevel` argument change which method is shown.

Reply:
True! I added the `filterwarnings` because without it, python wasn't showing a method at all! Why doesn't this show the code that raised the warning?

```python
def f():
    import snowflake.snowpark.modin.pandas as pd
f()
```

even if i put the warning inside a function:

```python

def f():
    warnings.warn(
        "Snowpark pandas is currently in Public Preview."
        + " See https://docs.snowflake.com/LIMITEDACCESS/snowpark-pandas for details.",
        stacklevel=1,
    )

f()
```

maybe it's because __init__.py isn't run via a function.